File: Arrays_And_Hashing/JavaSolutions/FirstUniqueCharacterInaString.java

package Arrays_And_Hashing.JavaSolutions;

class Solution {
    public int firstUniqChar(String s) {
        int[] freq = new int[26]; // Only lowercase English letters

        // Count frequencies
        for (int i = 0; i < s.length(); i++) {
            freq[s.charAt(i) - 'a']++;
        }

        // Find first unique character
        for (int i = 0; i < s.length(); i++) {
            if (freq[s.charAt(i) - 'a'] == 1) {
                return i;
            }
        }

        return -1; // No unique character found
    }
}

File: Arrays_And_Hashing/PythonSolutions/FirstUniqueCharacterInaString.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        
        freq = {}

        # Count frequencies
        for char in s:
            freq[char] = freq.get(char, 0) + 1

        # Find first unique character
        for i, char in enumerate(s):
            if freq[char] == 1:
                return i

        return -1  # No unique character found
